### PR TITLE
Fix(docs): Improve doc for DateInput

### PR DESCRIPTION
**Pull request by thien-do**
Please see the DateInput page in the deployment to review. This PR also includes some minor enhancements for the Doc in general, which are extracted to their own commits

### CODE REVIEW

File: core/src/date-input/date-input.tsx [hunk 1]
```diff
@@ -113,6 +113,20 @@ const getDisabledDays = (props: Props): Modifier => {
 	return days;
 };
 
+/**
+ * Date Inputs let users enter a date, by either typing or choosing from a
+ * pop-up calendar. They are an alternative to `<Input type="date"\>` that
```
Comment (ng-hai) [inline core/src/date-input/date-input.tsx:118]: You mean "There are.." right?

Comment (thien-do) [reply core/src/date-input/date-input.tsx:118]: No, I do mean "They are". "They" here mean the "Date Inputs", which are alternatives to the built in one.

But anyway, I think "They are an alternative" sounds wrong. Should be "They are alternatives to ..."

Comment (thien-do) [reply core/src/date-input/date-input.tsx]: @ng-hai can you check again? I've just updated it to be more clear

Comment (ng-hai) [reply core/src/date-input/date-input.tsx]: Done

**Review by ng-hai — APPROVED** [review]

**Pull request merged**